components/private_channel/rust/ffi/src/private_channel.cpp

```diff
@@ -0,0 +1,25 @@
+/* Copyright 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/private_channel/rust/ffi/src/private_channel.hpp"
+#include <iostream>
+
+extern "C" {
+#include "brave/components/private_channel/rust/ffi/src/lib.h"
+}
+
+namespace private_channel {
+
+  C_ResultChallenge start_challenge(
+    const char** input_ptr, int size, const uint8_t* server_pk) {
+      return client_start_challenge(input_ptr, size, server_pk);
+  }
+
+  C_ResultSecondRound second_round(
+    const uint8_t* enc_input_ptr, int input_size, const uint8_t*  sk) {
+      return client_second_round(enc_input_ptr, input_size, sk);
+  }
```

Comment:
The best FFIs are the ones where you can't tell the implementation is in a different language - this file would be a great place to create a single idiomatic C++ API so that types like the raw `C_ResultChallenge` and `const char**` are hidden from the rest of `brave-core`.

Reply:
- [ ] Refactor `private_channel.cpp` to be more cpp idiomatic